### DISCUSSION TITLE
MariaDB docker-compose.yml setup?

### DISCUSSION THREAD

**Discussion by Duke-LeTran** [nocodb/nocodb#1428] [Q&A]
Hello,

The simple docker quick starts work for me!! But unfortunately, I can't seem to configure the **docker-compose.yml** correctly, specifically for a **MariaDB** database. Would you be able to help me identify what might be incorrect my yml file?

## Relevant Points
* I created my own database called `app_nocodb` to store the metadata - is this okay? 
* I also added the `collation-server` flag per [code](https://github.com/nocodb/nocodb/issues/1313#issuecomment-1046625974) as recommended from @wingkwong referenced in the docs
* Additionally, I don't see `/usr/app` - do I need to generate these folders? Or do I need to [mount](https://docs.nocodb.com/getting-started/installation#node-app--docker) it (how do I do so)?
* Server OS: `Ubuntu 18.04.6 LTS x86_64`

## docker-compose.yml
* https://mariadb.com/kb/en/setting-up-a-lamp-stack-with-docker-compose/
* https://docs.nocodb.com/getting-started/installation#docker-compose

```
# docker-compose.yml
version: '2.1'

services:
  root_db:
    # https://mariadb.com/kb/en/setting-up-a-lamp-stack-with-docker-compose/
    image: mariadb:10.5.12 #mysql:5.7
    volumes:
      - db_data:/var/lib/mysql
#    below line shows how to change charset and collation
#    uncomment it if necessary
    command: --character-set-server=utf8mb4 --collation-server=utf8mb4_unicode_ci
    restart: always
    environment:
      MYSQL_ROOT_PASSWORD: changeme #password
      MYSQL_DATABASE: app_nocodb #root_db
      MYSQL_USER: app_nocodb #noco
      MYSQL_PASSWORD: changeme #password
    healthcheck:
      test: [ "CMD", "mysqladmin" ,"ping", "-h", "localhost" ]
      timeout: 20s
      retries: 10

  nocodb:
    depends_on:
      root_db:
        condition: service_healthy
    image: nocodb/nocodb:latest
    ports:
      - "8080:8080"
    restart: always
    volumes:
      - nc_data:/usr/app/data
    environment:
      #NC_DB: "mysql2://root_db:3306?u=noco&p=password&d=root_db"
      NC_DB: "mysql2://app_nocodb:3306?u=app_nocodb&password&d=nocodb" #maybe this is incorrect??
volumes:
  db_data: {}
  nc_data: {}
```
Please let me know if you have any advice.

**Comment by wingkwong** [comment]
It seems you didn't pass the password correctly.

```
NC_DB: "mysql2://app_nocodb:3306?u=app_nocodb&password&d=nocodb"
```

Please try the below one instead.

```
NC_DB: "mysql2://app_nocodb:3306?u=app_nocodb&p=password&d=nocodb"
```

As the credential may have stored in volume. Please remove it before retrying.

**Comment by Duke-LeTran** [comment]
Thank @wingkwong. Issue resolved on archived Discord general thread "docker-compose issue"
